# DENTIST R and cpp cpmparison


## Overview

DENTIST code is implemented in cpp `https://github.com/Yves-CHEN/DENTIST`. To incorporate it in the pecotmr package an R version is developed. 

Here we will test if the R version code have the same result of cpp version.

## Important note

DENTIST does not have a easy way to compile it in the system. Their paths in the makefile is quite messy and we need to install all dependencies and re-wrote the make file, which is quite time consuming. 

However, DENTIST provided an excutable file that can do the same things, but we cannot modify it so that we can only have input and outputs. Given that DENTIST involves randomness in splitting the region into S1 and S2 two parts and iterated this multiple times, so the numerical value of different attempts might be different. So here our strategy is to run DENTIST 10 times in each system, take the average for numerical things and compare if they are similar enough.

## Input

DENTIST the original program needs PLINK file as input, and also sumstat in COJO format (including columns `SNP A1 A2 freq b se p N`). Here we use the first round RSS_QC result as reference, found a region have outliers after allele QC. We use this region to verify the similarity of result.

+ PLINK file: `/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK_input_MWE`

+ Sumstat: `/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_MAF_filtered.txt`


Notes about inputs: DENTIST only allow region with 2000+ variables, also, should not include variants that have maf = 0 in PLINK. So here the plink file is already filtered by threshold 0.01.

## Details

### 1. MWE data preparation

For DENTIST in github https://github.com/Yves-CHEN/DENTIST version 1.3.0.0, it requires PLINK data, so here I prepared it.

In [217]:
library(tidyverse)
library(susieR)
library(plink2R)
library(pecotmr)
library(vroom)

# These variants exist in both plink file and original bellenguez sumstat data

sumstat = read_tsv("/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat.tsv")

# get variant list to extract from PLINK data
sumstat %>% pull(SNP) %>% 
    write.table("/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/MWE_snplist.txt", row.names = F, quote = F)

Rows: 2485 Columns: 8
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (3): SNP, A1, A2
dbl (5): freq, b, se, p, N

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [ ]:
rds$LD 

In [ ]:
sumstat

In [ ]:
./DENTIST_1.3.0.0 --bfile  /home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK --gwas-summary /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat.tsv --out D_3000

In [204]:
# extract the PLINK files from chromosome 1

plink2 --bfile /mnt/vast/hpc/csg/FunGen_xQTL/ROSMAP/Genotype/geno_by_chrom/ROSMAP_NIA_WGS.leftnorm.bcftools_qc.plink_qc.1 --extract /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/MWE_snplist.txt --make-bed --maf 0.01 --geno 0.01 --out ~/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK

PLINK v2.00a5LM 64-bit Intel (23 Sep 2023)     www.cog-genomics.org/plink/2.0/
(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK.log.
Options in effect:
  --bfile /mnt/vast/hpc/csg/FunGen_xQTL/ROSMAP/Genotype/geno_by_chrom/ROSMAP_NIA_WGS.leftnorm.bcftools_qc.plink_qc.1
  --extract /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/MWE_snplist.txt
  --geno 0.01
  --maf 0.01
  --make-bed
  --out /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK

Start time: Fri Mar 15 15:58:48 2024
515666 MiB RAM detected, ~467195 available; reserving 257833 MiB for main
workspace.
Allocated 81579 MiB successfully, after larger attempt(s) failed.
Using up to 48 threads (change this with --threads).
1153 samples (0 females, 0 males, 1153 ambiguous; 1153 founders) loaded from
/mnt/vast/hpc/csg/FunGen_xQTL/ROSMAP/Genotype/geno_by_chrom/ROSMAP_NIA_WGS.leftnorm.bcftools_qc.plink_qc.1.fam.



In [205]:
genotype = read_plink("~/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK")

In [214]:
# calculate LD matrix to run it in rCPP interface 

geno = read_plink("/home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK")$bed
LD = as.matrix(cor(geno))
write.table(LD, "/home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/LD.tsv", sep = "\t", quote = FALSE, row.names = TRUE, col.names = TRUE)

#########FINISHED############
#sumstat = read_delim("/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_MAF_filtered.txt")
LD = vroom("/home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/LD.tsv")[,-1]
LD = as.matrix(LD)

Warning message in fread("/home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/LD.tsv"):
“Detected 2485 column names but the data has 2486 columns (i.e. invalid file). Added 1 extra default column name for the first column which is guessed to be row names or an index. Use setnames() afterwards if this guess is not correct, or fix the file write command that created the file to create a valid file.”


In [218]:
LD

chr1:206510390_A_G chr1:206511644_A_C chr1:206512294_T_C
 [1,] 1.00000000         0.99409214         0.22390305        
 [2,] 0.99409214         1.00000000         0.22262588        
 [3,] 0.22390305         0.22262588         1.00000000        
 [4,] 0.90411284         0.90555517         0.20167758        
 [5,] 0.17337791         0.17534669         0.06443504        
 [6,] 0.43821645         0.43900313         0.40632115        
 [7,] 0.15516385         0.14809457         0.04647080        
 [8,] 0.08338565         0.07757592         0.01478903        
 [9,] 0.13202961         0.13340517         0.03450568        
[10,] 0.99494335         0.99240582         0.22278903        
[11,] 0.99661864         0.99237204         0.21906807        
[12,] 0.89385915         0.89533441         0.20089331        
[13,] 0.98732227         0.98139848         0.21960528        
[14,] 0.98657406         0.98240463         0.22091549        
[15,] 0.14689804         0.14123283         0.03344521        
[16,] 0.98732227         0.98139848         0.21960528        
[17,] 0.24699165         0.24531263         0.78751774        
[18,] 0.89354807         0.89334288         0.20156239        
[19,] 0.14698411         0.13983729         0.03365430        
[20,] 0.14887742         0.14161426         0.03206512        
[21,] 0.98326938         0.97911988         0.22242012        
[22,] 0.14984849         0.14252578         0.03125953        
[23,] 0.14887742         0.14161426         0.03206512        
[24,] 0.15613490         0.14900890         0.03365430        
[25,] 0.15083648         0.14345325         0.04280714        
[26,] 0.88877252         0.88688189         0.20273912        
[27,] 0.15083648         0.14345325         0.04280714        
[28,] 0.61216850         0.61047988         0.34836081        
[29,] 0.14689804         0.14123283         0.03344521        
[30,] 0.15712233         0.14993873         0.03286331        
[31,] ⋮                  ⋮                  ⋮                 
[32,]  0.001935122        0.0008194345      -0.03643271       
[33,] -0.005141850       -0.0062583087      -0.03545610       
[34,]  0.121768943        0.1232687562       0.02628885       
[35,]  0.014047598        0.0186160558       0.02307681       
[36,] -0.085218392       -0.0871021371      -0.05724594       
[37,]  0.121768943        0.1232687562       0.02628885       
[38,]  0.121768943        0.1232687562       0.02628885       
[39,]  0.121768943        0.1232687562       0.02628885       
[40,] -0.045121507       -0.0503199844      -0.03958824       
[41,] -0.044971921       -0.0501812873      -0.04028564       
[42,] -0.084591661       -0.0864849244      -0.04559803       
[43,]  0.121768943        0.1232687562       0.02628885       
[44,] -0.051874244       -0.0589025591      -0.04255314       
[45,]  0.121768943        0.1232687562       0.02628885       
[46,] -0.053549165       -0.0605923807      -0.04293545       
[47,]  0.121768943        0.1232687562       0.02628885       
[48,]  0.011649301        0.0133215445       0.03022610       
[49,]  0.023731048        0.0275577142       0.04562804       
[50,] -0.019458464       -0.0174999554      -0.04611464       
[51,]  0.121768943        0.1232687562       0.02628885       
[52,]  0.121768943        0.1232687562       0.02628885       
[53,]  0.121768943        0.1232687562       0.02628885       
[54,] -0.020632414       -0.0220252092      -0.02420090       
[55,]  0.121768943        0.1232687562       0.02628885       
[56,]  0.121768943        0.1232687562       0.02628885       
[57,]  0.121768943        0.1232687562       0.02628885       
[58,]  0.008728159        0.0076264783      -0.03737749       
[59,]  0.121768943        0.1232687562       0.02628885       
[60,]  0.002082977        0.0009814535      -0.03643698       
[61,]  0.121768943        0.1232687562       0.02628885       
      chr1:206512396_C_T chr1:206513188_C_T chr1:206514603_T_C
 [1,]  0.90411284         0.17337791 

4123 variants and 1153 samples

In [215]:
X = geno
EAF = c()
for(mm in 1:ncol(X)){
            EAF[mm] = sum(X[,mm])/(2*nrow(X))
}

sumstat$freq = EAF
write_tsv(sumstat, "/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat.tsv")


In [216]:
sumstat

SNP                 A1 A2 freq       b       se     p           N   
1    chr1:206510390_A_G  G  A  0.51691240 -0.0030 0.0082 0.3572371   1153
2    chr1:206511644_A_C  C  A  0.51474415 -0.0047 0.0082 0.2832645   1153
3    chr1:206512294_T_C  C  T  0.96010408  0.0069 0.0212 0.6275880   1153
4    chr1:206512396_C_T  T  C  0.46357329 -0.0025 0.0082 0.3802295   1153
5    chr1:206513188_C_T  T  C  0.03469211  0.0165 0.0238 0.7559322   1153
6    chr1:206514603_T_C  C  T  0.79011275  0.0010 0.0105 0.5379371   1153
7    chr1:206516658_C_T  T  C  0.03729402 -0.0070 0.0217 0.3735064   1153
8    chr1:206516810_C_T  T  C  0.01517780 -0.0239 0.0352 0.2485761   1153
9    chr1:206516811_G_A  A  G  0.01604510 -0.0152 0.0335 0.3250111   1153
10   chr1:206516953_C_T  T  C  0.51604510 -0.0031 0.0082 0.3526972   1153
11   chr1:206517450_T_C  C  T  0.51604510 -0.0025 0.0082 0.3802295   1153
12   chr1:206518744_G_T  T  G  0.46704250 -0.0016 0.0083 0.4235691   1153
13   chr1:206519300_T_G  G  T  0.52081526 -0.0016 0.0082 0.4226487   1153
14   chr1:206520410_A_C  C  A  0.52298352 -0.0034 0.0083 0.3410356   1153
15   chr1:206520435_C_T  T  C  0.01517780  0.0017 0.0342 0.5198223   1153
16   chr1:206520545_T_C  C  T  0.52081526 -0.0021 0.0082 0.3989378   1153
17   chr1:206520594_A_G  G  A  0.93668690 -0.0028 0.0172 0.4353416   1153
18   chr1:206520642_G_A  A  G  0.46790980 -0.0008 0.0083 0.4616072   1153
19   chr1:206522398_G_A  A  G  0.03686036 -0.0063 0.0225 0.3897388   1153
20   chr1:206524916_G_A  A  G  0.03599306 -0.0095 0.0230 0.3397874   1153
21   chr1:206525016_A_G  G  A  0.52471813 -0.0041 0.0083 0.3106616   1153
22   chr1:206525156_T_C  C  T  0.03555941 -0.0166 0.0234 0.2390376   1153
23   chr1:206525408_G_T  T  G  0.03599306 -0.0090 0.0230 0.3477861   1153
24   chr1:206529122_T_C  C  T  0.03686036 -0.0103 0.0220 0.3198273   1153
25   chr1:206530440_T_C  C  T  0.03512576 -0.0188 0.0236 0.2128387   1153
26   chr1:206530627_A_G  G  A  0.46921075 -0.0019 0.0083 0.4094672   1153
27   chr1:206531606_C_T  T  C  0.03512576 -0.0180 0.0236 0.2228176   1153
28   chr1:206532276_A_G  G  A  0.74067650 -0.0112 0.0097 0.1241191   1153
29   chr1:206532343_C_T  T  C  0.01517780 -0.0183 0.0343 0.2968342   1153
30   chr1:206532906_A_G  G  A  0.03642671 -0.0163 0.0224 0.2334052   1153
⋮    ⋮                   ⋮  ⋮  ⋮          ⋮       ⋮      ⋮           ⋮   
2456 chr1:207931491_A_G  G  A  0.98352125  0.0474 0.0274 0.958178341 1153
2457 chr1:207931511_C_T  T  C  0.98395490  0.0602 0.0280 0.984222393 1153
2458 chr1:207931631_C_T  T  C  0.01951431 -0.0320 0.0314 0.154075782 1153
2459 chr1:207931643_T_C  C  T  0.35862966 -0.0223 0.0084 0.003968220 1153
2460 chr1:207931971_T_C  C  T  0.96400694  0.0387 0.0207 0.969227890 1153
2461 chr1:207932265_A_G  G  A  0.01951431 -0.0320 0.0314 0.154075782 1153
2462 chr1:207932376_C_T  T  C  0.01951431 -0.0295 0.0315 0.174505864 1153
2463 chr1:207932567_A_G  G  A  0.01951431 -0.0326 0.0314 0.149584622 1153
2464 chr1:207933016_G_T  T  G  0.60494363  0.0278 0.0082 0.999650848 1153
2465 chr1:207933223_T_C  C  T  0.60407632  0.0283 0.0082 0.999720970 1153
2466 chr1:207934065_G_A  A  G  0.96357329  0.0383 0.0207 0.967860628 1153
2467 chr1:207934403_C_T  T  C  0.01951431 -0.0321 0.0314 0.153321124 1153
2468 chr1:207934647_C_T  T  C  0.59540330  0.0270 0.0082 0.999503818 1153
2469 chr1:207935537_A_G  G  A  0.01951431 -0.0325 0.0314 0.150327019 1153
2470 chr1:207935602_C_T  T  C  0.59496964  0.0271 0.0082 0.999524911 1153
2471 chr1:207935786_C_T  T  C  0.01951431 -0.0327 0.0314 0.148844676 1153
2472 chr1:207935787_G_A  A  G  0.17519514  0.0141 0.0110 0.900046783 1153
2473 chr1:207935840_G_A  A  G  0.30702515 -0.0238 0.0086 0.002824906 1153
2474 chr1:207935981_G_A  A  G  0.05160451  0.0006 0.0184 0.513006682 1153
2475 chr1:207936078_C_T  T  C  0.01951431 -0.0327 0.0314 0.148844676 1153
2476 chr1:207936120_G_A  A  G  0.01951431 -0.0327 0.0314 0.148844676 1153
2477 chr1:207936435_A_G  G  A  0.01951431 -0.0328 0.0314 0.148107179 1153
2478 chr1

In [200]:
LD[,1]


chr1:206011748_A_G 
        0.06745939

### 2. DENTIST  -- github compiled version implementation

In [ ]:
# DENTIST -- github compiled version

~/RSS_QC/DENTIST/DENTIST_1.3.0.0  --bfile  /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK_input_sep --gwas-summary /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/sumstat_sep_20.tsv --out DENTIST_40var

In [ ]:
# DENTIST -- github compiled version

~/RSS_QC/DENTIST/DENTIST_1.3.0.0  --bfile  /home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK_input_MWE --gwas-summary /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_MAF_updated.txt --out DENTIST_new

sumstat: /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/sumstat_sep_20.tsv
LD: /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/LD_sep_20.txt
PLINK: /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK_input_sep

In [20]:
wc -l ~/RSS_QC/DENTIST/DENTIST_result.DENTIST.short.txt

63 /home/hs3393/RSS_QC/DENTIST/DENTIST_result.DENTIST.short.txt


Here the row number of *.short.txt shows how many outliers are identified in this region and should be removed.

So for DENTIST compiled version, the **number of outlier is 63**. 

### 3. DENTIST -- Rcpp version

In [35]:
sumstat = read_delim("/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_MAF_filtered.txt") #%>% mutate(z = b / se)
LD = vroom("/home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/LD_MWE.tsv")[,-1]
LD = as.matrix(LD)
dentist_result = dentist(zScore = sumstat$z, LD = LD, nSample = 1153)

Rows: 4123 Columns: 8
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (3): SNP, A1, A2
dbl (5): freq, b, se, p, N

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in fread("/home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/LD_MWE.tsv"):
“Detected 4123 column names but the data has 4124 columns (i.e. invalid file). Added 1 extra default column name for the first column which is guessed to be row names or an index. Use setnames() afterwards if this guess is not correct, or fix the file write command that created the file to create a valid file.”


In [119]:
sumstat = read_delim("/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_MAF_filtered.txt") #%>% mutate(z = b / se)

Rows: 4123 Columns: 8
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (3): SNP, A1, A2
dbl (5): freq, b, se, p, N

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [139]:
dentist_result

imputed_z   rsq       corrected_z iter_to_correct is_problematic
1    -1.83233588 0.9843676   2.9014788 10              0             
2    -1.14057440 0.9652836  -2.7764632 10              0             
3    -1.15736728 0.9672809  -2.5731372 10              0             
4    -1.24256029 0.9885428  -3.3885340 10              0             
5     0.28931101 0.6342879  -0.2464973 10              0             
6    -1.70599901 0.9849100   1.7831310 10              0             
7    -1.47904789 0.9530842  -0.1171599 10              0             
8    -0.36542404 0.9731943  -8.3081095  1              0             
9    -1.95041772 0.8768238   2.8081253 10              0             
10   -1.28476532 0.5818886   1.9098788 10              0             
11   -0.95860919 0.8131786  -2.2440965 10              0             
12    0.24528981 0.7854401  -3.8970995 10              0             
13   -0.28882895 0.9901279   2.3334613 10              0             
14   -0.00744930 0.9729789  -1.5491997 10              0             
15    0.31030117 0.9762612  -2.4364239 10              0             
16   -0.37838707 0.9883213  -0.4492267 10              0             
17    0.30999226 0.6904111  -0.4370077 10              0             
18    0.66840403 0.9937082  -1.7986305 10              0             
19   -0.24317353 0.9915953  -3.0655672 10              0             
20    0.37122841 0.9939023   3.1914007 10              0             
21   -0.24317353 0.9915953  -3.4377073 10              0             
22   -0.35918560 0.9868379   1.2430316 10              0             
23   -0.37838707 0.9883213  -0.4681331 10              0             
24   -0.21238223 0.9976914  -0.2831857 10              0             
25    0.16993538 0.9955672  -0.1298459 10              0             
26   -1.72043716 0.6729591   1.9751315 10              0             
27    0.31227660 0.8810075   1.3968312 10              0             
28    0.34550130 0.9923208  -1.7275156 10              0             
29   -0.07206833 0.9769213   2.8995450 10              0             
30    0.38979743 0.9970972 -10.9145947  0              0             
⋮    ⋮           ⋮         ⋮           ⋮               ⋮             
4094  0.54264081 0.9923316 -1.7762422  10              0             
4095  0.00000000 0.0000000  0.0000000  10              0             
4096 -1.19816384 0.7086599  1.1496312  10              0             
4097  0.00000000 0.0000000  0.0000000  10              0             
4098  0.00000000 0.0000000  0.0000000  10              0             
4099  0.00000000 0.0000000  0.0000000  10              0             
4100 -0.33286165 0.9491033  1.5263802  10              0             
4101 -0.45823398 0.9731510  1.4403543  10              0             
4102  0.16910870 0.9927322  3.5424979  10              0             
4103  0.16910870 0.9927322  1.4132819  10              0             
4104  0.72495144 0.8509905  0.7125288  10              0             
4105  1.03485911 0.7202317 -3.5490881  10              0             
4106  0.00000000 0.0000000  0.0000000  10              0             
4107  0.66446784 0.9030072 -2.4657218  10              0             
4108  0.00000000 0.0000000  0.0000000  10              0             
4109  0.69532005 0.9844360 -6.0978355   0              0             
4110  0.37887875 0.9848035 -1.7399852  10              0             
4111  1.47696045 0.5513412 -1.2482461  10              0             
4112  0.00000000 0.0000000  0.0000000  10              0             
4113  0.82702227 0.9276249 -0.9192863  10              0             
4114  0.00000000 0.0000000  0.0000000  10              0             
4115  2.44407749 0.5229990 -3.8643301  10              0             
4116  0.00000000 0.0000000  0.0000000  10              0             
4117  0.00000000 0.0000000  0.0000000  10              0             
4118  3.18057161 0.6889265 -5.4515965   0              0             
4119 -1.1403919

In [36]:
dentist_result %>% filter(is_problematic > 0) %>% nrow()

[1] 1053

So for DENTIST RCPP version, the **number of outlier is 1053**. The results looks quite different... Take a look at the LD.

In [69]:
library(readr)
library(dplyr, warn.conflicts = FALSE)
library(vroom)
library(RcppArmadillo)
library(Rcpp)
source("/home/rd2972/software/pecotmr/R/run_dentist.R")
source("/home/rd2972/software/pecotmr/R/RcppExports.R")
sourceCpp("/home/hs3393/RSS_QC/pecotmr/src/dentist.cpp")
sourceCpp("/home/rd2972/software/pecotmr/src/RcppExports.cpp")
#dentist_LD_result = vroom("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/output_LD.txt")

Warning message in sourceCpp("/home/rd2972/software/pecotmr/src/RcppExports.cpp"):
“No Rcpp::export attributes or RCPP_MODULE declarations found in source”


In [55]:
LD

chr1:206011748_A_G chr1:206012565_A_G chr1:206012721_C_G
 [1,]  1.00000000         0.98500629         0.98215134       
 [2,]  0.98500629         1.00000000         0.99703144       
 [3,]  0.98215134         0.99703144         1.00000000       
 [4,]  0.97722319         0.98655338         0.98367739       
 [5,] -0.16241358        -0.16597212        -0.16673209       
 [6,]  0.98013632         0.98062367         0.97479826       
 [7,]  0.94805046         0.94822880         0.94244220       
 [8,]  0.94283218         0.95174415         0.94886559       
 [9,]  0.31266017         0.30599136         0.30414100       
[10,]  0.41408706         0.40531837         0.40288333       
[11,]  0.58251096         0.57531861         0.57433986       
[12,]  0.28935799         0.28270539         0.28087510       
[13,]  0.19779936         0.20217400         0.21003039       
[14,]  0.18323232         0.19661504         0.21322948       
[15,]  0.16698877         0.19635348         0.21189003       
[16,]  0.18697656         0.20005226         0.21642110       
[17,]  0.23869597         0.24286537         0.24135066       
[18,]  0.19243985         0.19602804         0.19463296       
[19,]  0.18697656         0.20005226         0.21642110       
[20,]  0.19243985         0.19602804         0.19463296       
[21,]  0.18697656         0.20005226         0.21642110       
[22,]  0.20366795         0.19873534         0.19738577       
[23,]  0.18697656         0.20005226         0.21642110       
[24,]  0.18323232         0.19661504         0.21322948       
[25,]  0.16789131         0.19809598         0.21403379       
[26,] -0.07010229        -0.07143634        -0.07171487       
[27,]  0.55363363         0.54699978         0.54499157       
[28,]  0.16789131         0.19809598         0.21403379       
[29,]  0.19646769         0.19169692         0.19039200       
[30,]  0.17285144         0.20344025         0.21957233       
[31,] ⋮                  ⋮                  ⋮                 
[32,] -0.0091603060      -0.0126553405      -0.013505590      
[33,] -0.0093453696      -0.0127679696      -0.013600339      
[34,]  0.0349953546       0.0286634938       0.027087489      
[35,] -0.0091603060      -0.0126553405      -0.013505590      
[36,] -0.0091603060      -0.0126553405      -0.013505590      
[37,]  0.0295317561       0.0159497229       0.017325303      
[38,]  0.0124788325       0.0175219957       0.013545644      
[39,]  0.0238330715       0.0101161669       0.011426403      
[40,] -0.0093453696      -0.0127679696      -0.013600339      
[41,] -0.0093453696      -0.0127679696      -0.013600339      
[42,]  0.0103892240      -0.0007561224       0.003150928      
[43,] -0.0063708208      -0.0046928718      -0.005669176      
[44,] -0.0125115125      -0.0094938946      -0.010831264      
[45,]  0.0042552123      -0.0014634494       0.003336064      
[46,] -0.0069343675      -0.0104613643      -0.011320984      
[47,] -0.0069343675      -0.0104613643      -0.011320984      
[48,] -0.0069343675      -0.0104613643      -0.011320984      
[49,]  0.0005898472      -0.0014138337      -0.001905311      
[50,] -0.0069343675      -0.0104613643      -0.011320984      
[51,] -0.0170794133      -0.0157594164      -0.013559374      
[52,] -0.0219637636      -0.0242452626      -0.024789252      
[53,] -0.0193532592      -0.0216382623      -0.022184913      
[54,] -0.0138176149      -0.0055390197      -0.004293321      
[55,] -0.0394362221      -0.0419444286      -0.042531913      
[56,]  0.0256967804       0.0225030342       0.021702552      
[57,] -0.0053929916      -0.0077023607      -0.008264625      
[58,] -0.0135394991      -0.0153967323      -0.015842559      
[59,] -0.0461785952      -0.0429995460      -0.043702051      
[60,] -0.0443735833      -0.0411619006      -0.041865722      
[61,] -0.0411658907      -0.0382699506      -0.039014423      
      chr1:206013858_A_G chr1:206014205_G_A chr1:206015713_G_A
 [1,]  0.97722319        -0.16241358 

In [166]:
original_plink = vroom("/mnt/vast/hpc/csg/FunGen_xQTL/ROSMAP/Genotype/geno_by_chrom/ROSMAP_NIA_WGS.leftnorm.bcftools_qc.plink_qc.1.bim")

In [54]:
result1 = vroom("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/per_iteration/dentist_Rcpp/20240308_version/DENTIST_Rcpp_output_01.txt")

V1          V2          V3         V4          V5         V6         
1     0.985010    0.997030   -0.165970   0.980620    0.9517400  0.3059900 
2     0.977220    0.983680   -0.165530   0.990970    0.9623700  0.3093300 
3     0.948050    0.942440   -0.078334   0.964930    0.9912700  0.2999800 
4     0.414090    0.402880   -0.056151   0.414770    0.3990600 -0.0349970 
5     0.582510    0.574340   -0.182270   0.589240    0.6041700  0.1733100 
6     0.289360    0.280880   -0.048627   0.290170    0.2728200  0.9004400 
7     0.197800    0.210030   -0.066790   0.198570    0.1854400  0.6333200 
8     0.183230    0.213230   -0.064954   0.184010    0.1714600  0.5908600 
9     0.186980    0.216420   -0.066252   0.187760    0.1750000  0.6072600 
10    0.203670    0.197390   -0.065482   0.204390    0.1912600  0.6430300 
11    0.186980    0.216420   -0.066252   0.187760    0.1750000  0.6072600 
12    0.183230    0.213230   -0.064954   0.184010    0.1714600  0.5908600 
13   -0.070102   -0.071715    0.375760  -0.068439   -0.0751770 -0.0351290 
14    0.553630    0.544990    0.448120   0.560930    0.5796500 -0.0552040 
15    0.196470    0.190390   -0.062798   0.197180    0.1844600  0.6368800 
16    0.172850    0.219570   -0.067530   0.191460    0.1784800  0.5745100 
17    0.183230    0.213230   -0.064954   0.184010    0.1714600  0.5908600 
18   -0.106950   -0.109250    0.532680  -0.104490   -0.1143100 -0.0335180 
19    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
20    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
21    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
22    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
23    0.195780    0.216480   -0.069344   0.205710    0.1921800  0.5899400 
24    0.201390    0.213280   -0.068077   0.202170    0.1888400  0.5985800 
25    0.195780    0.216480   -0.069344   0.205710    0.1921800  0.5899400 
26   -0.083472   -0.084728    0.360640  -0.081827   -0.0878720 -0.0334980 
27   -0.042468   -0.038344   -0.044900  -0.040446   -0.0082962 -0.0067722 
28    0.600050    0.596750    0.105730   0.609540    0.6185400  0.1349300 
29    0.591750    0.588410    0.089708   0.606890    0.5725100  0.1403300 
30    0.289490    0.297330    0.121350   0.297090    0.3109000 -0.0820300 
⋮    ⋮           ⋮           ⋮          ⋮           ⋮          ⋮          
2033 -0.02617300 -0.02911200 -0.0189420 -0.02436800 -0.0338530  0.00194470
2034 -0.04218500 -0.04634000 -0.0304290 -0.03957000 -0.0421160 -0.03011500
2035 -0.03122200 -0.02890300  0.0761340 -0.02564500 -0.0247160 -0.01611800
2036 -0.04218500 -0.04634000 -0.0304290 -0.03957000 -0.0421160 -0.03011500
2037 -0.03366700 -0.03872600 -0.0175730 -0.03568000 -0.0272390 -0.05695300
2038 -0.03379400 -0.03149500  0.0744000 -0.02817000 -0.0273780 -0.01695400
2039 -0.03300700 -0.03069600  0.0717960 -0.02738600 -0.0265470 -0.01671100
2040 -0.04218500 -0.04634000 -0.0304290 -0.03957000 -0.0421160 -0.03011500
2041 -0.01060900 -0.00912980  0.0713670 -0.00950030 -0.0020405  0.00203220
2042  0.00359900  0.00034147 -0.0099817  0.00520660 -0.0044106  0.02998500
2043 -0.02505600 -0.02356400  0.0465750 -0.03736000 -0.0300180  0.00876280
2044  0.04118400  0.03997900  0.0567980  0.05053700  0.0530240  0.01178700
2045 -0.00320020 -0.00747060 -0.0277660 -0.00099723 -0.0138220 -0.01951900
2046 -0.00916030 -0.01350600 -0.0195010 -0.00684750 -0.0081100 -0.02283700
2047  0.03499500  0.02708700 -0.0135120  0.03858100  0.0237290 -0.01523100
2048  0.01247900  0.01354600 -0.0363520  0.01296000  0.0164580 -0.03235800
2049 -0.00934540 -0.01360000 -0.0154410 -0.00707610 -0.0078823 -0.02120100
2050 -0.00934540 -0.01360000 -0.0154410 -0.00707610 -0.0078823 -0.02120100
2051  0.01038900  0.00315090  0.0148820  0.00345350 -0.0059086  0.02258200
2052 -0.00637080 -0.00566920 -0.0110040 -0.00382770 -0.0074890 -0.00030993
2053 -0.00693440 -0.01132100 -0.0140630 -0.00462750 -0.0062140 -0.02362200
2054 -0.00693440 -0.01132100 -

ERROR: Error in parse(text = x, srcfile = src): <text>:1:150: unexpected input
1: result2 = fread("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/per_iteration/dentist_Rcpp/20240308_version/DENTIST_Rcpp_output_11.txt"）
                                                                                                                                                         ^


In [56]:
result1 = vroom("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/per_iteration/dentist_Rcpp/20240308_version/DENTIST_Rcpp_output_02.txt")

In [73]:
dentist_result = dentist(zScore = sumstat$z[2001:4001], LD = LD[2001:4001, 2001:4001], nSample = 1153)

ERROR: Error in value[[3L]](cond): Adjusted rsq_eigen value exceeding 1: 1.000443


In [52]:
dim(dentist_LD_result)
head(dentist_LD_result)

[1] 2062 2061

LD_it   for     iteration :       V5      V6        V7        V8       
1 0.98501 0.99703 -0.165970 0.98062 0.95174  0.305990  0.196350  0.242870
2 0.97722 0.98368 -0.165530 0.99097 0.96237  0.309330  0.182120  0.245680
3 0.94805 0.94244 -0.078334 0.96493 0.99127  0.299980  0.174640  0.237840
4 0.41409 0.40288 -0.056151 0.41477 0.39906 -0.034997 -0.049078 -0.025289
5 0.58251 0.57434 -0.182270 0.58924 0.60417  0.173310  0.068187  0.168260
6 0.28936 0.28088 -0.048627 0.29017 0.27282  0.900440  0.708410 -0.039013
  V9          V10       ⋯ V2052      V2053       V2054      V2055      
1  1.9603e-01  0.200050 ⋯  0.0159500  0.01011600 -0.0094939 -1.4634e-03
2  1.9891e-01  0.184900 ⋯  0.0252130  0.01956100 -0.0130840  4.3563e-03
3  1.9103e-01  0.177670 ⋯  0.0142890  0.00809080 -0.0169470  9.1581e-05
4 -3.2538e-02 -0.046423 ⋯  0.0057649 -0.00058517 -0.0405290 -1.2314e-03
5  8.4623e-02  0.077967 ⋯ -0.0131740 -0.02404500  0.0247050 -6.2777e-03
6 -7.5352e-05  0.724340 ⋯ -0.0075276 -0.01285700 -0.0218220 -3.2347e-03
  V2056      V2057      V2058     V2059       V2060     V2061    
1 -0.0104610 -0.0104610 -0.024245 -0.00553900 -0.041944 -0.043000
2 -0.0073216 -0.0073216 -0.022075 -0.01147100 -0.039437 -0.040075
3 -0.0035575 -0.0035575 -0.027552 -0.00036976 -0.045728 -0.047427
4 -0.0481700 -0.0481700 -0.034190  0.01815900 -0.046387 -0.032131
5  0.0038312  0.0038312 -0.047474  0.04000800 -0.034784 -0.031634
6 -0.0328830 -0.0328830 -0.023340  0.04111700 -0.016986 -0.014725

Ignore the column names. The dimension of LD is not correct yet, also the diagnol elements are not equal to 1.

In [58]:
DENTIST_compile = vroom("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/DENTIST_Rcpp_output.txt")

Warning message in fread("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/DENTIST_Rcpp_output.txt"):
“Detected 4 column names but the data has 2061 columns (i.e. invalid file). Added 2057 extra default column names at the end.”
Warning message in fread("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/DENTIST_Rcpp_output.txt"):
“Stopped early on line 2064. Expected 2061 fields but found 0. Consider fill=TRUE and comment.char=. First discarded non-empty line: <<LD_it for iteration :>>”


In [174]:
EAF

[1] 0.16565481 0.16912402 0.16999133 0.16608846 0.08976583 0.16348656
   [7] 0.17389419 0.17649610 0.01951431 0.03295750 0.36470078 0.02385082
  [13] 0.01561145 0.01561145 0.01777971 0.01604510 0.01474415 0.01821336
  [19] 0.01604510 0.01821336 0.01604510 0.01517780 0.01604510 0.01561145
  [25] 0.01691240 0.01474415 0.23677363 0.01691240 0.01431049 0.01647875
  [31] 0.01561145 0.01561145 0.01604510 0.03165655 0.01604510 0.01604510
  [37] 0.01604510 0.01604510 0.01561145 0.01604510 0.01561145 0.01647875
  [43] 0.01604510 0.01604510 0.01647875 0.01344319 0.01517780 0.28360798
  [49] 0.02948829 0.02601908 0.22333044 0.18516912 0.19254120 0.03252385
  [55] 0.31613183 0.32393755 0.34301821 0.10884649 0.01170859 0.04553339
  [61] 0.03816132 0.01431049 0.05941023 0.03035559 0.08889853 0.10234172
  [67] 0.10754553 0.10234172 0.10277537 0.10234172 0.09583695 0.50346921
  [73] 0.09150043 0.01517780 0.08889853 0.50173461 0.08889853 0.24631396
  [79] 0.01561145 0.08846487 0.01691240 0.08889853 0.02515178 0.12619254
  [85] 0.09410234 0.08889853 0.02732003 0.09366869 0.09366869 0.09410234
  [91] 0.11795317 0.08889853 0.09366869 0.02515178 0.34388552 0.08803122
  [97] 0.09366869 0.08803122 0.08933218 0.08803122 0.08933218 0.08803122
 [103] 0.08889853 0.07155247 0.06027754 0.16392021 0.14787511 0.25151778
 [109] 0.12185603 0.12228968 0.12228968 0.15698179 0.15698179 0.12228968
 [115] 0.12228968 0.11188205 0.12185603 0.12922810 0.12055507 0.11144840
 [121] 0.01040763 0.06895056 0.01040763 0.15698179 0.13183001 0.02601908
 [127] 0.10797918 0.01084128 0.01040763 0.12836080 0.02601908 0.06721596
 [133] 0.04466609 0.03599306 0.04596704 0.15351258 0.15264527 0.13443192
 [139] 0.15351258 0.15307892 0.10841284 0.15307892 0.13529922 0.15307892
 [145] 0.02601908 0.12575889 0.12532524 0.12922810 0.12575889 0.03989592
 [151] 0.05333912 0.03339115 0.10190807 0.10320902 0.10190807 0.10234172
 [157] 0.10190807 0.10190807 0.10190807 0.10190807 0.10190807 0.10190807
 [163] 0.10190807 0.10581093 0.10190807 0.10190807 0.10190807 0.08586297
 [169] 0.10234172 0.10190807 0.10190807 0.10190807 0.10190807 0.10407632
 [175] 0.10407632 0.10407632 0.10537728 0.10537728 0.10537728 0.10450997
 [181] 0.09453599 0.10407632 0.10667823 0.10494363 0.10537728 0.10537728
 [187] 0.13790113 0.10537728 0.10537728 0.10581093 0.10537728 0.10537728
 [193] 0.10581093 0.10537728 0.10537728 0.14180399 0.01951431 0.10537728
 [199] 0.10537728 0.10537728 0.08933218 0.13790113 0.01821336 0.07935820
 [205] 0.03122290 0.02428448 0.04813530 0.07979185 0.08499566 0.02038161
 [211] 0.08022550 0.03122290 0.02645273 0.03122290 0.08109280 0.03122290
 [217] 0.02645273 0.03078925 0.02645273 0.04163053 0.07892454 0.02645273
 [223] 0.15958369 0.04163053 0.02645273 0.07849089 0.02688638 0.07805724
 [229] 0.37467476 0.03122290 0.01474415 0.01257589 0.04076323 0.04163053
 [235] 0.01821336 0.02298352 0.01647875 0.90806592 0.01257589 0.01257589
 [241] 0.02385082 0.01257589 0.01257589 0.41587164 0.06461405 0.03382480
 [247] 0.02948829 0.04856895 0.04856895 0.01864701 0.51214224 0.02818734
 [253] 0.02688638 0.02775369 0.47398092 0.02385082 0.02038161 0.01431049
 [259] 0.12619254 0.01387684 0.01431049 0.47224631 0.02948829 0.01474415
 [265] 0.01431049 0.03599306 0.02948829 0.01431049 0.01387684 0.02081526
 [271] 0.02948829 0.02948829 0.02948829 0.01474415 0.01474415 0.01431049
 [277] 0.01951431 0.01387684 0.01431049 0.02948829 0.01474415 0.01474415
 [283] 0.01864701 0.01474415 0.01474415 0.01431049 0.01474415 0.01387684
 [289] 0.01474415 0.02948829 0.01431049 0.01431049 0.01474415 0.01474415
 [295] 0.03035559 0.01431049 0.02948829 0.01431049 0.02948829 0.01474415
 [301] 0.01474415 0.01431049 0.01474415 0.49913270 0.01387684 0.02948829
 [307] 0.01431049 0.01387684 0.01431049 0.04466609 0.23807459 0.01431049
 [313] 0.01431049 0.01431049 0.05117086 0.02948829 0.01474415 0.01431049
 [319] 0.01431049 0.02948829 0.02948829 0.01431049 0.01431049 0.02905464
 [325] 0.02905464 0.49956635 0.01431049 0.23850824 0.01431049

In [115]:
sumstat

SNP                     A1 A2     freq       b       se     p         
1    chr1:206011748_A_G      G  A      0.16565481 -0.0169 0.0115 0.07083977
2    chr1:206012565_A_G      G  A      0.16912402 -0.0189 0.0114 0.04866936
3    chr1:206012721_C_G      G  C      0.16999133 -0.0185 0.0114 0.05231533
4    chr1:206013858_A_G      G  A      0.16608846 -0.0183 0.0114 0.05421795
5    chr1:206014205_G_A      A  G      0.08976583  0.0023 0.0164 0.55576636
6    chr1:206015713_G_A      A  G      0.16348656 -0.0171 0.0115 0.06851315
7    chr1:206016996_A_T      T  A      0.17389419 -0.0170 0.0113 0.06623601
8    chr1:206018106_T_C      C  T      0.17649610 -0.0195 0.0113 0.04220396
9    chr1:206018145_C_T      T  C      0.01951431 -0.0357 0.0370 0.16730625
10   chr1:206018463_T_C      C  T      0.03295750 -0.0013 0.0261 0.48013752
11   chr1:206018722_T_C      C  T      0.36470078 -0.0189 0.0098 0.02689204
12   chr1:206022249_T_C      C  T      0.02385082 -0.0482 0.0309 0.05939524
13   chr1:206023800_C_T      T  C      0.01561145 -0.0020 0.0351 0.47728054
14   chr1:206023974_G_C      C  G      0.01561145 -0.0092 0.0351 0.39661899
15   chr1:206024034_A_G      G  A      0.01777971 -0.0022 0.0338 0.47405167
16   chr1:206025435_T_C      C  T      0.01604510 -0.0149 0.0349 0.33471366
17   chr1:206025897_G_A      A  G      0.01474415  0.0026 0.0389 0.52664468
18   chr1:206026330_G_A      A  G      0.01821336  0.0143 0.0272 0.70046393
19   chr1:206027447_G_C      C  G      0.01604510 -0.0184 0.0351 0.30006397
20   chr1:206027728_C_T      T  C      0.01821336  0.0170 0.0274 0.73251526
21   chr1:206029363_A_G      G  A      0.01604510 -0.0201 0.0360 0.28830839
22   chr1:206029868_G_A      A  G      0.01517780 -0.0081 0.0374 0.41426880
23   chr1:206030218_G_A      A  G      0.01604510 -0.0151 0.0352 0.33396988
24   chr1:206031382_G_C      C  G      0.01561145 -0.0080 0.0354 0.41060510
25   chr1:206032030_C_T      T  C      0.01691240  0.0055 0.0341 0.56406763
26   chr1:206032219_T_C      C  T      0.01474415 -0.0195 0.0330 0.27729067
27   chr1:206034003_T_C      C  T      0.23677363  0.0081 0.0102 0.78643653
28   chr1:206034513_G_A      A  G      0.01691240  0.0066 0.0340 0.57695812
29   chr1:206034840_C_T      T  C      0.01431049  0.0147 0.0399 0.64372035
30   chr1:206034918_T_G      G  T      0.01647875 -0.0068 0.0343 0.42142445
⋮    ⋮                       ⋮  ⋮      ⋮          ⋮       ⋮      ⋮         
4094 chr1:208448615_C_T      T  C      0.03729402  0.0084 0.0217 0.6506577 
4095 chr1:208448650_G_A      A  G      0.03599306  0.0068 0.0221 0.6208418 
4096 chr1:208448992_C_A      A  C      0.09323504 -0.0093 0.0161 0.2817537 
4097 chr1:208449524_A_G      G  A      0.03729402  0.0055 0.0214 0.6014142 
4098 chr1:208449747_A_G      G  A      0.03729402  0.0054 0.0215 0.5991559 
4099 chr1:208449897_C_A      A  C      0.34778838 -0.0044 0.0091 0.3143645 
4100 chr1:208450130_T_G      G  T      0.42237641  0.0001 0.0087 0.5045854 
4101 chr1:208450277_C_A      A  C      0.35602775 -0.0020 0.0090 0.4120704 
4102 chr1:208450310_CACG_C   C  CACG   0.03599306  0.0106 0.0225 0.6812193 
4103 chr1:208450528_C_T      T  C      0.03599306  0.0064 0.0221 0.6139361 
4104 chr1:208450938_A_G      G  A      0.16955768  0.0114 0.0114 0.8413447 
4105 chr1:208451406_C_T      T  C      0.04466609 -0.0171 0.0203 0.1997920 
4106 chr1:208452386_G_A      A  G      0.08239376 -0.0010 0.0148 0.4730649 
4107 chr1:208452730_A_G      G  A      0.46877710 -0.0009 0.0087 0.4588036 
4108 chr1:208452930_C_T      T  C      0.03816132 -0.0006 0.0213 0.4887637 
4109 chr1:208453740_A_G      G  A      0.03816132 -0.0014 0.0214 0.4739196 
4110 chr1:208453895_CAAAGA_C C  CAAAGA 0.03816132  0.0036 0.0219 0.5652854 
4111 chr1:208454302_T_C      C  T      0.01170859  0.0298 0.0465 0.7391932 
4112 chr1:208454412_C_T      T  C      0.03816132 -0.0011 0.0216 0.4796923 
4113 chr1:208455240_A_G      G  A      0.10667823  0.0080 0.0138 0.7189449 
4114 chr1:208455523_C_T      T  C      0.01908066  0.

In [175]:
sumstat$freq = EAF

In [153]:
sumstat

SNP                     A1 A2     freq       b       se     p         
1    chr1:206011748_A_G      G  A      0.16565481 -0.0169 0.0115 0.07083977
2    chr1:206012565_A_G      G  A      0.16912402 -0.0189 0.0114 0.04866936
3    chr1:206012721_C_G      G  C      0.16999133 -0.0185 0.0114 0.05231533
4    chr1:206013858_A_G      G  A      0.16608846 -0.0183 0.0114 0.05421795
5    chr1:206014205_G_A      A  G      0.08976583  0.0023 0.0164 0.55576636
6    chr1:206015713_G_A      A  G      0.16348656 -0.0171 0.0115 0.06851315
7    chr1:206016996_A_T      T  A      0.17389419 -0.0170 0.0113 0.06623601
8    chr1:206018106_T_C      C  T      0.17649610 -0.0195 0.0113 0.04220396
9    chr1:206018145_C_T      T  C      0.01951431 -0.0357 0.0370 0.16730625
10   chr1:206018463_T_C      C  T      0.03295750 -0.0013 0.0261 0.48013752
11   chr1:206018722_T_C      C  T      0.36470078 -0.0189 0.0098 0.02689204
12   chr1:206022249_T_C      C  T      0.02385082 -0.0482 0.0309 0.05939524
13   chr1:206023800_C_T      T  C      0.01561145 -0.0020 0.0351 0.47728054
14   chr1:206023974_G_C      C  G      0.01561145 -0.0092 0.0351 0.39661899
15   chr1:206024034_A_G      G  A      0.01777971 -0.0022 0.0338 0.47405167
16   chr1:206025435_T_C      C  T      0.01604510 -0.0149 0.0349 0.33471366
17   chr1:206025897_G_A      A  G      0.01474415  0.0026 0.0389 0.52664468
18   chr1:206026330_G_A      A  G      0.01821336  0.0143 0.0272 0.70046393
19   chr1:206027447_G_C      C  G      0.01604510 -0.0184 0.0351 0.30006397
20   chr1:206027728_C_T      T  C      0.01821336  0.0170 0.0274 0.73251526
21   chr1:206029363_A_G      G  A      0.01604510 -0.0201 0.0360 0.28830839
22   chr1:206029868_G_A      A  G      0.01517780 -0.0081 0.0374 0.41426880
23   chr1:206030218_G_A      A  G      0.01604510 -0.0151 0.0352 0.33396988
24   chr1:206031382_G_C      C  G      0.01561145 -0.0080 0.0354 0.41060510
25   chr1:206032030_C_T      T  C      0.01691240  0.0055 0.0341 0.56406763
26   chr1:206032219_T_C      C  T      0.01474415 -0.0195 0.0330 0.27729067
27   chr1:206034003_T_C      C  T      0.23677363  0.0081 0.0102 0.78643653
28   chr1:206034513_G_A      A  G      0.01691240  0.0066 0.0340 0.57695812
29   chr1:206034840_C_T      T  C      0.01431049  0.0147 0.0399 0.64372035
30   chr1:206034918_T_G      G  T      0.01647875 -0.0068 0.0343 0.42142445
⋮    ⋮                       ⋮  ⋮      ⋮          ⋮       ⋮      ⋮         
4094 chr1:208448615_C_T      T  C      0.03729402  0.0084 0.0217 0.6506577 
4095 chr1:208448650_G_A      A  G      0.03599306  0.0068 0.0221 0.6208418 
4096 chr1:208448992_C_A      A  C      0.09323504 -0.0093 0.0161 0.2817537 
4097 chr1:208449524_A_G      G  A      0.03729402  0.0055 0.0214 0.6014142 
4098 chr1:208449747_A_G      G  A      0.03729402  0.0054 0.0215 0.5991559 
4099 chr1:208449897_C_A      A  C      0.34778838 -0.0044 0.0091 0.3143645 
4100 chr1:208450130_T_G      G  T      0.42237641  0.0001 0.0087 0.5045854 
4101 chr1:208450277_C_A      A  C      0.35602775 -0.0020 0.0090 0.4120704 
4102 chr1:208450310_CACG_C   C  CACG   0.03599306  0.0106 0.0225 0.6812193 
4103 chr1:208450528_C_T      T  C      0.03599306  0.0064 0.0221 0.6139361 
4104 chr1:208450938_A_G      G  A      0.16955768  0.0114 0.0114 0.8413447 
4105 chr1:208451406_C_T      T  C      0.04466609 -0.0171 0.0203 0.1997920 
4106 chr1:208452386_G_A      A  G      0.08239376 -0.0010 0.0148 0.4730649 
4107 chr1:208452730_A_G      G  A      0.46877710 -0.0009 0.0087 0.4588036 
4108 chr1:208452930_C_T      T  C      0.03816132 -0.0006 0.0213 0.4887637 
4109 chr1:208453740_A_G      G  A      0.03816132 -0.0014 0.0214 0.4739196 
4110 chr1:208453895_CAAAGA_C C  CAAAGA 0.03816132  0.0036 0.0219 0.5652854 
4111 chr1:208454302_T_C      C  T      0.01170859  0.0298 0.0465 0.7391932 
4112 chr1:208454412_C_T      T  C      0.03816132 -0.0011 0.0216 0.4796923 
4113 chr1:208455240_A_G      G  A      0.10667823  0.0080 0.0138 0.7189449 
4114 chr1:208455523_C_T      T  C      0.01908066  0.

In [94]:
sumstat %>% filter(freq < 0.5)

SNP                     A1 A2     freq       b       se     p         
1    chr1:206011748_A_G      G  A      0.16565481 -0.0169 0.0115 0.07083977
2    chr1:206012565_A_G      G  A      0.16912402 -0.0189 0.0114 0.04866936
3    chr1:206012721_C_G      G  C      0.16999133 -0.0185 0.0114 0.05231533
4    chr1:206013858_A_G      G  A      0.16608846 -0.0183 0.0114 0.05421795
5    chr1:206014205_G_A      A  G      0.08976583  0.0023 0.0164 0.55576636
6    chr1:206015713_G_A      A  G      0.16348656 -0.0171 0.0115 0.06851315
7    chr1:206016996_A_T      T  A      0.17389419 -0.0170 0.0113 0.06623601
8    chr1:206018106_T_C      C  T      0.17649610 -0.0195 0.0113 0.04220396
9    chr1:206018145_C_T      T  C      0.01951431 -0.0357 0.0370 0.16730625
10   chr1:206018463_T_C      C  T      0.03295750 -0.0013 0.0261 0.48013752
11   chr1:206018722_T_C      C  T      0.36470078 -0.0189 0.0098 0.02689204
12   chr1:206022249_T_C      C  T      0.02385082 -0.0482 0.0309 0.05939524
13   chr1:206023800_C_T      T  C      0.01561145 -0.0020 0.0351 0.47728054
14   chr1:206023974_G_C      C  G      0.01561145 -0.0092 0.0351 0.39661899
15   chr1:206024034_A_G      G  A      0.01777971 -0.0022 0.0338 0.47405167
16   chr1:206025435_T_C      C  T      0.01604510 -0.0149 0.0349 0.33471366
17   chr1:206025897_G_A      A  G      0.01474415  0.0026 0.0389 0.52664468
18   chr1:206026330_G_A      A  G      0.01821336  0.0143 0.0272 0.70046393
19   chr1:206027447_G_C      C  G      0.01604510 -0.0184 0.0351 0.30006397
20   chr1:206027728_C_T      T  C      0.01821336  0.0170 0.0274 0.73251526
21   chr1:206029363_A_G      G  A      0.01604510 -0.0201 0.0360 0.28830839
22   chr1:206029868_G_A      A  G      0.01517780 -0.0081 0.0374 0.41426880
23   chr1:206030218_G_A      A  G      0.01604510 -0.0151 0.0352 0.33396988
24   chr1:206031382_G_C      C  G      0.01561145 -0.0080 0.0354 0.41060510
25   chr1:206032030_C_T      T  C      0.01691240  0.0055 0.0341 0.56406763
26   chr1:206032219_T_C      C  T      0.01474415 -0.0195 0.0330 0.27729067
27   chr1:206034003_T_C      C  T      0.23677363  0.0081 0.0102 0.78643653
28   chr1:206034513_G_A      A  G      0.01691240  0.0066 0.0340 0.57695812
29   chr1:206034840_C_T      T  C      0.01431049  0.0147 0.0399 0.64372035
30   chr1:206034918_T_G      G  T      0.01647875 -0.0068 0.0343 0.42142445
⋮    ⋮                       ⋮  ⋮      ⋮          ⋮       ⋮      ⋮         
4094 chr1:208448615_C_T      T  C      0.03729402  0.0084 0.0217 0.6506577 
4095 chr1:208448650_G_A      A  G      0.03599306  0.0068 0.0221 0.6208418 
4096 chr1:208448992_C_A      A  C      0.09323504 -0.0093 0.0161 0.2817537 
4097 chr1:208449524_A_G      G  A      0.03729402  0.0055 0.0214 0.6014142 
4098 chr1:208449747_A_G      G  A      0.03729402  0.0054 0.0215 0.5991559 
4099 chr1:208449897_C_A      A  C      0.34778838 -0.0044 0.0091 0.3143645 
4100 chr1:208450130_T_G      G  T      0.42237641  0.0001 0.0087 0.5045854 
4101 chr1:208450277_C_A      A  C      0.35602775 -0.0020 0.0090 0.4120704 
4102 chr1:208450310_CACG_C   C  CACG   0.03599306  0.0106 0.0225 0.6812193 
4103 chr1:208450528_C_T      T  C      0.03599306  0.0064 0.0221 0.6139361 
4104 chr1:208450938_A_G      G  A      0.16955768  0.0114 0.0114 0.8413447 
4105 chr1:208451406_C_T      T  C      0.04466609 -0.0171 0.0203 0.1997920 
4106 chr1:208452386_G_A      A  G      0.08239376 -0.0010 0.0148 0.4730649 
4107 chr1:208452730_A_G      G  A      0.46877710 -0.0009 0.0087 0.4588036 
4108 chr1:208452930_C_T      T  C      0.03816132 -0.0006 0.0213 0.4887637 
4109 chr1:208453740_A_G      G  A      0.03816132 -0.0014 0.0214 0.4739196 
4110 chr1:208453895_CAAAGA_C C  CAAAGA 0.03816132  0.0036 0.0219 0.5652854 
4111 chr1:208454302_T_C      C  T      0.01170859  0.0298 0.0465 0.7391932 
4112 chr1:208454412_C_T      T  C      0.03816132 -0.0011 0.0216 0.4796923 
4113 chr1:208455240_A_G      G  A      0.10667823  0.0080 0.0138 0.7189449 
4114 chr1:208455523_C_T      T  C      0.01908066  0.

In [176]:
write_tsv(sumstat, "/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_MAF_updated.txt")

In [ ]:
./DENTIST 

In [97]:
LD

chr1:206011748_A_G chr1:206012565_A_G chr1:206012721_C_G
 [1,]  1.00000000         0.98500629         0.98215134       
 [2,]  0.98500629         1.00000000         0.99703144       
 [3,]  0.98215134         0.99703144         1.00000000       
 [4,]  0.97722319         0.98655338         0.98367739       
 [5,] -0.16241358        -0.16597212        -0.16673209       
 [6,]  0.98013632         0.98062367         0.97479826       
 [7,]  0.94805046         0.94822880         0.94244220       
 [8,]  0.94283218         0.95174415         0.94886559       
 [9,]  0.31266017         0.30599136         0.30414100       
[10,]  0.41408706         0.40531837         0.40288333       
[11,]  0.58251096         0.57531861         0.57433986       
[12,]  0.28935799         0.28270539         0.28087510       
[13,]  0.19779936         0.20217400         0.21003039       
[14,]  0.18323232         0.19661504         0.21322948       
[15,]  0.16698877         0.19635348         0.21189003       
[16,]  0.18697656         0.20005226         0.21642110       
[17,]  0.23869597         0.24286537         0.24135066       
[18,]  0.19243985         0.19602804         0.19463296       
[19,]  0.18697656         0.20005226         0.21642110       
[20,]  0.19243985         0.19602804         0.19463296       
[21,]  0.18697656         0.20005226         0.21642110       
[22,]  0.20366795         0.19873534         0.19738577       
[23,]  0.18697656         0.20005226         0.21642110       
[24,]  0.18323232         0.19661504         0.21322948       
[25,]  0.16789131         0.19809598         0.21403379       
[26,] -0.07010229        -0.07143634        -0.07171487       
[27,]  0.55363363         0.54699978         0.54499157       
[28,]  0.16789131         0.19809598         0.21403379       
[29,]  0.19646769         0.19169692         0.19039200       
[30,]  0.17285144         0.20344025         0.21957233       
[31,] ⋮                  ⋮                  ⋮                 
[32,] -0.0091603060      -0.0126553405      -0.013505590      
[33,] -0.0093453696      -0.0127679696      -0.013600339      
[34,]  0.0349953546       0.0286634938       0.027087489      
[35,] -0.0091603060      -0.0126553405      -0.013505590      
[36,] -0.0091603060      -0.0126553405      -0.013505590      
[37,]  0.0295317561       0.0159497229       0.017325303      
[38,]  0.0124788325       0.0175219957       0.013545644      
[39,]  0.0238330715       0.0101161669       0.011426403      
[40,] -0.0093453696      -0.0127679696      -0.013600339      
[41,] -0.0093453696      -0.0127679696      -0.013600339      
[42,]  0.0103892240      -0.0007561224       0.003150928      
[43,] -0.0063708208      -0.0046928718      -0.005669176      
[44,] -0.0125115125      -0.0094938946      -0.010831264      
[45,]  0.0042552123      -0.0014634494       0.003336064      
[46,] -0.0069343675      -0.0104613643      -0.011320984      
[47,] -0.0069343675      -0.0104613643      -0.011320984      
[48,] -0.0069343675      -0.0104613643      -0.011320984      
[49,]  0.0005898472      -0.0014138337      -0.001905311      
[50,] -0.0069343675      -0.0104613643      -0.011320984      
[51,] -0.0170794133      -0.0157594164      -0.013559374      
[52,] -0.0219637636      -0.0242452626      -0.024789252      
[53,] -0.0193532592      -0.0216382623      -0.022184913      
[54,] -0.0138176149      -0.0055390197      -0.004293321      
[55,] -0.0394362221      -0.0419444286      -0.042531913      
[56,]  0.0256967804       0.0225030342       0.021702552      
[57,] -0.0053929916      -0.0077023607      -0.008264625      
[58,] -0.0135394991      -0.0153967323      -0.015842559      
[59,] -0.0461785952      -0.0429995460      -0.043702051      
[60,] -0.0443735833      -0.0411619006      -0.041865722      
[61,] -0.0411658907      -0.0382699506      -0.039014423      
      chr1:206013858_A_G chr1:206014205_G_A chr1:206015713_G_A
 [1,]  0.97722319        -0.16241358 

In [219]:
sumstat

SNP                 A1 A2 freq       b       se     p           N   
1    chr1:206510390_A_G  G  A  0.51691240 -0.0030 0.0082 0.3572371   1153
2    chr1:206511644_A_C  C  A  0.51474415 -0.0047 0.0082 0.2832645   1153
3    chr1:206512294_T_C  C  T  0.96010408  0.0069 0.0212 0.6275880   1153
4    chr1:206512396_C_T  T  C  0.46357329 -0.0025 0.0082 0.3802295   1153
5    chr1:206513188_C_T  T  C  0.03469211  0.0165 0.0238 0.7559322   1153
6    chr1:206514603_T_C  C  T  0.79011275  0.0010 0.0105 0.5379371   1153
7    chr1:206516658_C_T  T  C  0.03729402 -0.0070 0.0217 0.3735064   1153
8    chr1:206516810_C_T  T  C  0.01517780 -0.0239 0.0352 0.2485761   1153
9    chr1:206516811_G_A  A  G  0.01604510 -0.0152 0.0335 0.3250111   1153
10   chr1:206516953_C_T  T  C  0.51604510 -0.0031 0.0082 0.3526972   1153
11   chr1:206517450_T_C  C  T  0.51604510 -0.0025 0.0082 0.3802295   1153
12   chr1:206518744_G_T  T  G  0.46704250 -0.0016 0.0083 0.4235691   1153
13   chr1:206519300_T_G  G  T  0.52081526 -0.0016 0.0082 0.4226487   1153
14   chr1:206520410_A_C  C  A  0.52298352 -0.0034 0.0083 0.3410356   1153
15   chr1:206520435_C_T  T  C  0.01517780  0.0017 0.0342 0.5198223   1153
16   chr1:206520545_T_C  C  T  0.52081526 -0.0021 0.0082 0.3989378   1153
17   chr1:206520594_A_G  G  A  0.93668690 -0.0028 0.0172 0.4353416   1153
18   chr1:206520642_G_A  A  G  0.46790980 -0.0008 0.0083 0.4616072   1153
19   chr1:206522398_G_A  A  G  0.03686036 -0.0063 0.0225 0.3897388   1153
20   chr1:206524916_G_A  A  G  0.03599306 -0.0095 0.0230 0.3397874   1153
21   chr1:206525016_A_G  G  A  0.52471813 -0.0041 0.0083 0.3106616   1153
22   chr1:206525156_T_C  C  T  0.03555941 -0.0166 0.0234 0.2390376   1153
23   chr1:206525408_G_T  T  G  0.03599306 -0.0090 0.0230 0.3477861   1153
24   chr1:206529122_T_C  C  T  0.03686036 -0.0103 0.0220 0.3198273   1153
25   chr1:206530440_T_C  C  T  0.03512576 -0.0188 0.0236 0.2128387   1153
26   chr1:206530627_A_G  G  A  0.46921075 -0.0019 0.0083 0.4094672   1153
27   chr1:206531606_C_T  T  C  0.03512576 -0.0180 0.0236 0.2228176   1153
28   chr1:206532276_A_G  G  A  0.74067650 -0.0112 0.0097 0.1241191   1153
29   chr1:206532343_C_T  T  C  0.01517780 -0.0183 0.0343 0.2968342   1153
30   chr1:206532906_A_G  G  A  0.03642671 -0.0163 0.0224 0.2334052   1153
⋮    ⋮                   ⋮  ⋮  ⋮          ⋮       ⋮      ⋮           ⋮   
2456 chr1:207931491_A_G  G  A  0.98352125  0.0474 0.0274 0.958178341 1153
2457 chr1:207931511_C_T  T  C  0.98395490  0.0602 0.0280 0.984222393 1153
2458 chr1:207931631_C_T  T  C  0.01951431 -0.0320 0.0314 0.154075782 1153
2459 chr1:207931643_T_C  C  T  0.35862966 -0.0223 0.0084 0.003968220 1153
2460 chr1:207931971_T_C  C  T  0.96400694  0.0387 0.0207 0.969227890 1153
2461 chr1:207932265_A_G  G  A  0.01951431 -0.0320 0.0314 0.154075782 1153
2462 chr1:207932376_C_T  T  C  0.01951431 -0.0295 0.0315 0.174505864 1153
2463 chr1:207932567_A_G  G  A  0.01951431 -0.0326 0.0314 0.149584622 1153
2464 chr1:207933016_G_T  T  G  0.60494363  0.0278 0.0082 0.999650848 1153
2465 chr1:207933223_T_C  C  T  0.60407632  0.0283 0.0082 0.999720970 1153
2466 chr1:207934065_G_A  A  G  0.96357329  0.0383 0.0207 0.967860628 1153
2467 chr1:207934403_C_T  T  C  0.01951431 -0.0321 0.0314 0.153321124 1153
2468 chr1:207934647_C_T  T  C  0.59540330  0.0270 0.0082 0.999503818 1153
2469 chr1:207935537_A_G  G  A  0.01951431 -0.0325 0.0314 0.150327019 1153
2470 chr1:207935602_C_T  T  C  0.59496964  0.0271 0.0082 0.999524911 1153
2471 chr1:207935786_C_T  T  C  0.01951431 -0.0327 0.0314 0.148844676 1153
2472 chr1:207935787_G_A  A  G  0.17519514  0.0141 0.0110 0.900046783 1153
2473 chr1:207935840_G_A  A  G  0.30702515 -0.0238 0.0086 0.002824906 1153
2474 chr1:207935981_G_A  A  G  0.05160451  0.0006 0.0184 0.513006682 1153
2475 chr1:207936078_C_T  T  C  0.01951431 -0.0327 0.0314 0.148844676 1153
2476 chr1:207936120_G_A  A  G  0.01951431 -0.0327 0.0314 0.148844676 1153
2477 chr1:207936435_A_G  G  A  0.01951431 -0.0328 0.0314 0.148107179 1153
2478 chr1

In [59]:
DENTIST_compile

LD_it       for         iteration  :           V5         V6         
1     0.985010    0.997030   -0.165970   0.980620    0.9517400  0.3059900 
2     0.977220    0.983680   -0.165530   0.990970    0.9623700  0.3093300 
3     0.948050    0.942440   -0.078334   0.964930    0.9912700  0.2999800 
4     0.414090    0.402880   -0.056151   0.414770    0.3990600 -0.0349970 
5     0.582510    0.574340   -0.182270   0.589240    0.6041700  0.1733100 
6     0.289360    0.280880   -0.048627   0.290170    0.2728200  0.9004400 
7     0.197800    0.210030   -0.066790   0.198570    0.1854400  0.6333200 
8     0.183230    0.213230   -0.064954   0.184010    0.1714600  0.5908600 
9     0.186980    0.216420   -0.066252   0.187760    0.1750000  0.6072600 
10    0.203670    0.197390   -0.065482   0.204390    0.1912600  0.6430300 
11    0.186980    0.216420   -0.066252   0.187760    0.1750000  0.6072600 
12    0.183230    0.213230   -0.064954   0.184010    0.1714600  0.5908600 
13   -0.070102   -0.071715    0.375760  -0.068439   -0.0751770 -0.0351290 
14    0.553630    0.544990    0.448120   0.560930    0.5796500 -0.0552040 
15    0.196470    0.190390   -0.062798   0.197180    0.1844600  0.6368800 
16    0.172850    0.219570   -0.067530   0.191460    0.1784800  0.5745100 
17    0.183230    0.213230   -0.064954   0.184010    0.1714600  0.5908600 
18   -0.106950   -0.109250    0.532680  -0.104490   -0.1143100 -0.0335180 
19    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
20    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
21    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
22    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
23    0.195780    0.216480   -0.069344   0.205710    0.1921800  0.5899400 
24    0.201390    0.213280   -0.068077   0.202170    0.1888400  0.5985800 
25    0.195780    0.216480   -0.069344   0.205710    0.1921800  0.5899400 
26   -0.083472   -0.084728    0.360640  -0.081827   -0.0878720 -0.0334980 
27   -0.042468   -0.038344   -0.044900  -0.040446   -0.0082962 -0.0067722 
28    0.600050    0.596750    0.105730   0.609540    0.6185400  0.1349300 
29    0.591750    0.588410    0.089708   0.606890    0.5725100  0.1403300 
30    0.289490    0.297330    0.121350   0.297090    0.3109000 -0.0820300 
⋮    ⋮           ⋮           ⋮          ⋮           ⋮          ⋮          
2033 -0.02617300 -0.02911200 -0.0189420 -0.02436800 -0.0338530  0.00194470
2034 -0.04218500 -0.04634000 -0.0304290 -0.03957000 -0.0421160 -0.03011500
2035 -0.03122200 -0.02890300  0.0761340 -0.02564500 -0.0247160 -0.01611800
2036 -0.04218500 -0.04634000 -0.0304290 -0.03957000 -0.0421160 -0.03011500
2037 -0.03366700 -0.03872600 -0.0175730 -0.03568000 -0.0272390 -0.05695300
2038 -0.03379400 -0.03149500  0.0744000 -0.02817000 -0.0273780 -0.01695400
2039 -0.03300700 -0.03069600  0.0717960 -0.02738600 -0.0265470 -0.01671100
2040 -0.04218500 -0.04634000 -0.0304290 -0.03957000 -0.0421160 -0.03011500
2041 -0.01060900 -0.00912980  0.0713670 -0.00950030 -0.0020405  0.00203220
2042  0.00359900  0.00034147 -0.0099817  0.00520660 -0.0044106  0.02998500
2043 -0.02505600 -0.02356400  0.0465750 -0.03736000 -0.0300180  0.00876280
2044  0.04118400  0.03997900  0.0567980  0.05053700  0.0530240  0.01178700
2045 -0.00320020 -0.00747060 -0.0277660 -0.00099723 -0.0138220 -0.01951900
2046 -0.00916030 -0.01350600 -0.0195010 -0.00684750 -0.0081100 -0.02283700
2047  0.03499500  0.02708700 -0.0135120  0.03858100  0.0237290 -0.01523100
2048  0.01247900  0.01354600 -0.0363520  0.01296000  0.0164580 -0.03235800
2049 -0.00934540 -0.01360000 -0.0154410 -0.00707610 -0.0078823 -0.02120100
2050 -0.00934540 -0.01360000 -0.0154410 -0.00707610 -0.0078823 -0.02120100
2051  0.01038900  0.00315090  0.0148820  0.00345350 -0.0059086  0.02258200
2052 -0.00637080 -0.00566920 -0.0110040 -0.00382770 -0.0074890 -0.00030993
2053 -0.00693440 -0.01132100 -0.0140630 -0.00462750 -0.0062140 -0.02362200
2054 -0.00693440 -0.01132100 -

In [28]:
kk = vroom("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/per_iteration/dentist_original/4213_variants/LD_separate/LD_it_output_20.txt")

In [29]:
head(kk)

V1         V2          V3           V4          V5         V6        
1  0.6759160 -0.05131040 -1.92395e-03 -0.04470260  0.0505993 -0.0797282
2 -0.0286956  0.02563550  1.87835e-01 -0.03293410  0.0358906  0.0264595
3 -0.0742423  0.18604700  7.14334e-02  0.03134780 -0.0447789  0.1843570
4 -0.1909020  0.00375349 -1.68635e-03 -0.06660320  0.0394499 -0.0277484
5  0.8078360 -0.04763070 -7.06678e-05 -0.00322825  0.0854091 -0.0509672
6 -0.0531262  0.00703471  2.20129e-02  0.22299300 -0.0183365  0.0137763
  V7         V8          V9          V10         ⋯ V973       V974       
1  0.0436569  0.01908220  0.32164800  0.05074640 ⋯  0.1378230  0.11253900
2 -0.0358579 -0.04175240 -0.03144050 -0.03414390 ⋯  0.0164742  0.00802881
3 -0.1723980 -0.02332210 -0.04275870  0.00418691 ⋯ -0.0185382 -0.00656875
4  0.0172595 -0.00700894 -0.34048000 -0.05132800 ⋯ -0.0600390 -0.06478690
5  0.0533767  0.01319020  0.22029300  0.07152070 ⋯  0.0398345  0.00410843
6  0.0501553  0.02820910  0.00702352  0.07164150 ⋯ -0.0555523  0.04412450
  V975        V976        V977       V978        V979       V980      
1  0.03901070 -0.01396100 -0.0256802 -0.08519900  0.1151590 -0.2119800
2  0.00357216  0.02387510  0.0708860 -0.03225810 -0.0545845  0.0235798
3  0.01116950 -0.15475400  0.0525748  0.01008800 -0.0784397  0.0335985
4 -0.05054220 -0.00102061 -0.0511662  0.09739920 -0.0980070 -0.0827689
5  0.04491120  0.01159600 -0.0482848 -0.09297800  0.0922414 -0.2087910
6  0.09947470 -0.04475670  0.1023430  0.00352642 -0.0719310  0.0556175
  V981        V982      
1  0.04571330 -0.1430800
2 -0.00985897 -0.0141408
3 -0.07618340  0.0346303
4 -0.04171430  0.1791850
5  0.06020950 -0.1235290
6 -0.02644690 -0.0668466

In [26]:
k2 = vroom("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/per_iteration/dentist_Rcpp/4213_variants/DENTIST_Rcpp_output_01.txt")

In [27]:
sumstat: /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/sumstat_sep_20.tsv
LD: /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/LD_sep_20.txt
PLINK: /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK_input_sep

V1          V2          V3         V4          V5         V6         
1     0.985010    0.997030   -0.165970   0.980620    0.9517400  0.3059900 
2     0.977220    0.983680   -0.165530   0.990970    0.9623700  0.3093300 
3     0.948050    0.942440   -0.078334   0.964930    0.9912700  0.2999800 
4     0.414090    0.402880   -0.056151   0.414770    0.3990600 -0.0349970 
5     0.582510    0.574340   -0.182270   0.589240    0.6041700  0.1733100 
6     0.289360    0.280880   -0.048627   0.290170    0.2728200  0.9004400 
7     0.197800    0.210030   -0.066790   0.198570    0.1854400  0.6333200 
8     0.183230    0.213230   -0.064954   0.184010    0.1714600  0.5908600 
9     0.186980    0.216420   -0.066252   0.187760    0.1750000  0.6072600 
10    0.203670    0.197390   -0.065482   0.204390    0.1912600  0.6430300 
11    0.186980    0.216420   -0.066252   0.187760    0.1750000  0.6072600 
12    0.183230    0.213230   -0.064954   0.184010    0.1714600  0.5908600 
13   -0.070102   -0.071715    0.375760  -0.068439   -0.0751770 -0.0351290 
14    0.553630    0.544990    0.448120   0.560930    0.5796500 -0.0552040 
15    0.196470    0.190390   -0.062798   0.197180    0.1844600  0.6368800 
16    0.172850    0.219570   -0.067530   0.191460    0.1784800  0.5745100 
17    0.183230    0.213230   -0.064954   0.184010    0.1714600  0.5908600 
18   -0.106950   -0.109250    0.532680  -0.104490   -0.1143100 -0.0335180 
19    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
20    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
21    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
22    0.177960    0.216420   -0.066252   0.187760    0.1750000  0.5825300 
23    0.195780    0.216480   -0.069344   0.205710    0.1921800  0.5899400 
24    0.201390    0.213280   -0.068077   0.202170    0.1888400  0.5985800 
25    0.195780    0.216480   -0.069344   0.205710    0.1921800  0.5899400 
26   -0.083472   -0.084728    0.360640  -0.081827   -0.0878720 -0.0334980 
27   -0.042468   -0.038344   -0.044900  -0.040446   -0.0082962 -0.0067722 
28    0.600050    0.596750    0.105730   0.609540    0.6185400  0.1349300 
29    0.591750    0.588410    0.089708   0.606890    0.5725100  0.1403300 
30    0.289490    0.297330    0.121350   0.297090    0.3109000 -0.0820300 
⋮    ⋮           ⋮           ⋮          ⋮           ⋮          ⋮          
2033 -0.02617300 -0.02911200 -0.0189420 -0.02436800 -0.0338530  0.00194470
2034 -0.04218500 -0.04634000 -0.0304290 -0.03957000 -0.0421160 -0.03011500
2035 -0.03122200 -0.02890300  0.0761340 -0.02564500 -0.0247160 -0.01611800
2036 -0.04218500 -0.04634000 -0.0304290 -0.03957000 -0.0421160 -0.03011500
2037 -0.03366700 -0.03872600 -0.0175730 -0.03568000 -0.0272390 -0.05695300
2038 -0.03379400 -0.03149500  0.0744000 -0.02817000 -0.0273780 -0.01695400
2039 -0.03300700 -0.03069600  0.0717960 -0.02738600 -0.0265470 -0.01671100
2040 -0.04218500 -0.04634000 -0.0304290 -0.03957000 -0.0421160 -0.03011500
2041 -0.01060900 -0.00912980  0.0713670 -0.00950030 -0.0020405  0.00203220
2042  0.00359900  0.00034147 -0.0099817  0.00520660 -0.0044106  0.02998500
2043 -0.02505600 -0.02356400  0.0465750 -0.03736000 -0.0300180  0.00876280
2044  0.04118400  0.03997900  0.0567980  0.05053700  0.0530240  0.01178700
2045 -0.00320020 -0.00747060 -0.0277660 -0.00099723 -0.0138220 -0.01951900
2046 -0.00916030 -0.01350600 -0.0195010 -0.00684750 -0.0081100 -0.02283700
2047  0.03499500  0.02708700 -0.0135120  0.03858100  0.0237290 -0.01523100
2048  0.01247900  0.01354600 -0.0363520  0.01296000  0.0164580 -0.03235800
2049 -0.00934540 -0.01360000 -0.0154410 -0.00707610 -0.0078823 -0.02120100
2050 -0.00934540 -0.01360000 -0.0154410 -0.00707610 -0.0078823 -0.02120100
2051  0.01038900  0.00315090  0.0148820  0.00345350 -0.0059086  0.02258200
2052 -0.00637080 -0.00566920 -0.0110040 -0.00382770 -0.0074890 -0.00030993
2053 -0.00693440 -0.01132100 -0.0140630 -0.00462750 -0.0062140 -0.02362200
2054 -0.00693440 -0.01132100 -

In [220]:
rds = readRDS("~/RSS_QC/toy_example_zRdiscrep.rds")

In [224]:
var_names = vroom("/home/rd2972/private_data/20240300_rss_qc_imputation/DENTIST/per_iteration/dentist_original/4213_variants/out.DENTIST.full.txt")

In [234]:
var_name = var_names %>% pull(V1) %>% str_replace("chr", "") %>% str_replace("_", ":") %>% str_replace("_", ":")

In [239]:
var_name[2000] %in% colnames(rds$LD)

[1] FALSE

In [238]:
colnames(rds$LD)

[1] "1:158934025:C:T"                                                                                                                        
    [2] "1:158935661:G:A"                                                                                                                        
    [3] "1:158935773:A:T"                                                                                                                        
    [4] "1:158936157:CCATTAACTCAT:C"                                                                                                             
    [5] "1:158936329:C:A"                                                                                                                        
    [6] "1:158936330:G:A"                                                                                                                        
    [7] "1:158936487:G:C"                                                                                                                        
    [8] "1:158936692:T:C"                                                                                                                        
    [9] "1:158936765:A:C"                                                                                                                        
   [10] "1:158936987:A:G"                                                                                                                        
   [11] "1:158937449:T:C"                                                                                                                        
   [12] "1:158937633:T:C"                                                                                                                        
   [13] "1:158937772:C:T"                                                                                                                        
   [14] "1:158937787:A:T"                                                                                                                        
   [15] "1:158937804:G:A"                                                                                                                        
   [16] "1:158937810:G:C"                                                                                                                        
   [17] "1:158937961:A:C"                                                                                                                        
   [18] "1:158938022:G:A"                                                                                                                        
   [19] "1:158938246:C:T"                                                                                                                        
   [20] "1:158939096:C:T"                                                                                                                        
   [21] "1:158939180:C:T"                                                                                                                        
   [22] "1:158939374:C:A"                                                                                                                        
   [23] "1:158940259:A:G"                                                                                                                        
   [24] "1:158940386:AAG:A"                                                                                                                      
   [25] "1:158940404:G:A"                                                                                                                        
   [26] "1:158940507:T:C"                                                                                                                        
   [27] "1:158940600:A:T"                                                                                                                        
   [28] "1:158940606:AT:A"                                    

In [236]:
rds$LD[var_name, var_name]

ERROR: Error in rds$LD[var_name, var_name]: subscript out of bounds


In [240]:
sumstat_4k = vroom("~/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_4k.tsv")

In [242]:
sumstat_4k %>% pull(plink_variant_id) %>% 
    write.table("/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/4k_snplist.txt", row.names = F, quote = F)

In [ ]:
# extract the PLINK files from chromosome 1

plink2 --bfile /mnt/vast/hpc/csg/FunGen_xQTL/ROSMAP/Genotype/geno_by_chrom/ROSMAP_NIA_WGS.leftnorm.bcftools_qc.plink_qc.1 --extract /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/4k_snplist.txt --make-bed --maf 0.01 --geno 0.01 --out ~/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK_4k

In [243]:
# calculate LD matrix to run it in rCPP interface 

geno = read_plink("/home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK_4k")$bed
LD = as.matrix(cor(geno))
write.table(LD, "/home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/LD_4k.tsv", sep = "\t", quote = FALSE, row.names = TRUE, col.names = TRUE)

In [245]:
X = geno
EAF = c()
for(mm in 1:ncol(X)){
            EAF[mm] = sum(X[,mm])/(2*nrow(X))
}

In [246]:
sumstat_4k$maf = EAF

In [248]:
sumstat_4k %>% mutate(variant_id = paste0("chr", chrom, ":", pos, "_", A2, "_", A1))  %>% 
    mutate(pvalue = pnorm(beta / se)) %>%
    select(variant_id, A1, A2, maf, beta, se,  pvalue ,n_case) %>% rename(SNP = variant_id,
                                                                        freq = maf, b = beta,
                                                                        p = pvalue, N = n_case) %>%
    mutate(N = 1153) %>%
    write_tsv("~/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_4k.tsv")

In [283]:
sumstat_4k[730,]$pos - sumstat_4k[1,]$pos

[1] 606506

In [ ]:
./DENTIST_1.3.0.0 --bfile /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK_4k --gwas-summary /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_4k.tsv --out D_4k_8iter --no-missing-genotype  --wind-dist 2000000  --dup-threshold 1.0 --iteration-num 8

In [293]:
vroom("~/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_4k.tsv")[c(743:4123),] %>% pull(SNP) %>% 
    write.table("/home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/MWE_snplist_4k_window2.txt", row.names = F, quote = F)

In [ ]:
# extract the PLINK files from chromosome 1

plink2 --bfile /mnt/vast/hpc/csg/FunGen_xQTL/ROSMAP/Genotype/geno_by_chrom/ROSMAP_NIA_WGS.leftnorm.bcftools_qc.plink_qc.1 --extract /home/hs3393/RSS_QC/pecotmr/data/RSS_QC_MWE/MWE_snplist_4k_window2.txt --make-bed --maf 0.01 --geno 0.01 --out ~/RSS_QC/pecotmr/data/RSS_QC_MWE/PLINK_4k_window2

In [286]:
write.table(LD[2:4123, 2:4123], "/home/hs3393//RSS_QC/pecotmr/data/RSS_QC_MWE/LD_4k_even.tsv", sep = "\t", quote = FALSE, row.names = TRUE, col.names = TRUE)

In [292]:
vroom("~/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_4k.tsv")[c(743:4123),] %>% write_tsv("~/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_4k_window2.tsv")

In [289]:
(4123 - 2746) / 4123

[1] 0.3339801

In [296]:
2449520 / 4

[1] 612380

In [336]:
vroom("~/RSS_QC/pecotmr/data/RSS_QC_MWE/Sumstat_4k.tsv") %>% separate(SNP, into = c("chr", "pos"), sep = ":") %>%
separate(pos,into = c("pos", "var"), sep = "_") %>% mutate(pos = as.numeric(pos)) %>%
mutate(chr = str_replace(chr, "^chr", "")) %>% mutate(z = b / se, chr = as.numeric(chr))  %>%
select(chr, pos, A1, A2, z) %>% write_tsv("~/RSS_QC/pecotmr/data/RSS_QC_MWE/sumstat_pecotmr_format.tsv")

Warning message:
“Expected 2 pieces. Additional pieces discarded in 4123 rows [1, 2, 3, 4, 5, 6,
7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].”


In [308]:
(sumstat$pos[4123] - sumstat$pos[1])/4

[1] 612378.5

In [332]:
sumstat$pos[1407] - sumstat$pos[1]

sumstat$pos[515] - sumstat$pos[1]

sumstat$pos[1407] - sumstat$pos[515]

sumstat$pos[2126] - sumstat$pos[1407]

sumstat$pos[3181] - sumstat$pos[2126]


[1] 998002

[1] 498014

[1] 499988

[1] 495696

[1] 498734